test/test_sparse_csr.py

```diff
@@ -371,33 +369,48 @@ def test_csr_matvec(self, device, dtype):
             with self.assertRaisesRegex(RuntimeError, "mv: expected"):
                 csr.matmul(bad_vec)
 
-    @onlyCPU
     @dtypes(torch.double)
     def test_mm(self, device, dtype):
         def test_shape(di, dj, dk, nnz):
-            x = self.genSparseCSRTensor((di, dj), nnz, device=device, dtype=dtype, index_dtype=torch.int32)
-            t = torch.randn(di, dk, dtype=dtype, device=device)
-            y = torch.randn(dj, dk, dtype=dtype, device=device)
-            alpha = random.random()
-            beta = random.random()
-
-            # res = beta * t  + alpha * (x @ y)
-            res = torch.addmm(t, x, y, beta=beta, alpha=alpha)
-            expected = torch.addmm(t, x.to_dense(), y, beta=beta, alpha=alpha)
-            self.assertEqual(res, expected)
+            for index_dtype in [torch.int32, torch.int64]:
+                x = self.genSparseCSRTensor((di, dj), nnz, device=device, dtype=dtype, index_dtype=index_dtype)
+                t = torch.randn(di, dk, dtype=dtype, device=device)
+                y = torch.randn(dj, dk, dtype=dtype, device=device)
+                alpha = random.random()
+                beta = random.random()
 
-            res = torch.addmm(t, x, y)
-            expected = torch.addmm(t, x.to_dense(), y)
-            self.assertEqual(res, expected)
+                # res = beta * t  + alpha * (x @ y)
+                res = torch.addmm(t, x, y, beta=beta, alpha=alpha)
+                expected = torch.addmm(t, x.to_dense(), y, beta=beta, alpha=alpha)
+                self.assertEqual(res, expected)
 
-            res = torch.mm(x, y)
-            expected = torch.mm(x.to_dense(), y)
-            self.assertEqual(res, expected)
+                res = torch.addmm(t, x, y)
+                expected = torch.addmm(t, x.to_dense(), y)
+                self.assertEqual(res, expected)
+
+                res = torch.mm(x, y)
```

Comment:
add tests that make sure error is raised with device mismatch. Also, various discontiguous and/or broadcasted dense inputs should be tested

Reply:
Added: Test with discontinuous and/or broadcasted dense inputs. 

Can't add: Tests that make sure error is raised with device mismatch. This neither works with COO tensors. I create an issue for this here:   #59058

Reply:
I'm confused, you have device checks in your implementation - do they not throw an error?

Reply:
Yes, It throws an error. That's why I maintained the device check in this implementation. 
What I am saying is If I disabled these checks  and trust in the 
automatically generated checks it will fail as it reported here:   #59058

Reply:
So you can actually add a test that assertRaises on device mismatch? If autogeneration is not working, then please don't disable the checks in your implmenetation, and add tests, so that when autogeneration is fixed it can easily be tested.

Reply:
Ok I will add these test with current checks.